packages/engine/src/datastore/table/proxy.rs

```diff
@@ -62,7 +63,7 @@ impl<K: Batch> Clone for BatchReadProxy<K> {
         // SAFETY: Since a BatchReadProxy already exists, the existing BatchReadProxy
         //         must already have the shared lock, so no writer can currently have
         //         the lock, so it must be possible to take the shared lock again.
-        let locked = unsafe { self.arc.raw() }.try_lock_shared();
+        let locked = unsafe { RwLock::raw(&self.arc).try_lock_shared() };
         assert!(locked, "Clone BatchReadProxy");
```

Comment:
I know you didn't write it but this seems like a strange message for an assert

Reply:
I adjusted the assertion message, good catch!